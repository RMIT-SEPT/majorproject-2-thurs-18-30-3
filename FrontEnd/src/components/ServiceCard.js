import React from 'react'
import {Link} from 'react-router-dom'

import '../containers/App.css'

//Displays a particular service in a small, granular card
function ServiceCard({service}) {
  return (
    <Link to={`/services/${service.id}`} style={{textDecoration: 'none'}}>
      <button className="service-card" role="cell">
        <div className="card-content">
          <img src="not-found.png" alt={service.name} />
          <span>{service.name}</span>
          <hr />
          <p>{service.description}</p>
        </div>
      </button>
    </Link>
  )
}
export default ServiceCard
