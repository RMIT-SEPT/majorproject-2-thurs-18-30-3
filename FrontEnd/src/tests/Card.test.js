import React from 'react'

import ServiceCard from '../components/ServiceCard'
import {StaticRouter} from 'react-router'
import renderer from 'react-test-renderer'
import {render, screen, fireEvent} from '@testing-library/react'

import '@testing-library/jest-dom/extend-expect'

//Snapshot test
test('Test Card Renders', () => {
  const dummyService = {id: 1, name: 'TEST BUTTON', description: 'this is a test', photo: 'not-found.png'}

  const card = renderer
    .create(
      <StaticRouter>
        <ServiceCard service={dummyService} />
      </StaticRouter>
    )
    .toJSON()
  expect(card).toMatchSnapshot()
})

//Check that card displays desired content
test('Test Card Contents', () => {
  const dummyService = {id: 1, name: 'TEST BUTTON', description: 'this is a test', photo: 'not-found.png'}
  const container = render(
    <StaticRouter>
      <ServiceCard service={dummyService} />
    </StaticRouter>
  )

  expect(screen.getByRole('cell')).toHaveTextContent('TEST BUTTON')
  expect(screen.getByRole('img')).toHaveAttribute('src', 'not-found.png')
})

//Test that click does not change card
test('Test Card On Click', () => {
  const dummyService = {id: 1, name: 'TEST BUTTON', description: 'this is a test', photo: 'not-found.png'}
  const container = render(
    <StaticRouter>
      <ServiceCard service={dummyService} />
    </StaticRouter>
  )

  fireEvent.click(screen.getByRole('cell'))
  expect(container).toMatchSnapshot()
})
