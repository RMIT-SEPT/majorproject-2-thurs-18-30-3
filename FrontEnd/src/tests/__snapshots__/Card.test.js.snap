// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card On Click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/services/1"
        style="text-decoration: none;"
      >
        <button
          class="service-card"
          role="cell"
        >
          <div
            class="card-content"
          >
            <img
              alt="TEST BUTTON"
              src="not-found.png"
            />
            <span>
              TEST BUTTON
            </span>
            <hr />
            <p>
              this is a test
            </p>
          </div>
        </button>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/services/1"
      style="text-decoration: none;"
    >
      <button
        class="service-card"
        role="cell"
      >
        <div
          class="card-content"
        >
          <img
            alt="TEST BUTTON"
            src="not-found.png"
          />
          <span>
            TEST BUTTON
          </span>
          <hr />
          <p>
            this is a test
          </p>
        </div>
      </button>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Card Renders 1`] = `
<a
  href="/services/1"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <button
    className="service-card"
    role="cell"
  >
    <div
      className="card-content"
    >
      <img
        alt="TEST BUTTON"
        src="not-found.png"
      />
      <span>
        TEST BUTTON
      </span>
      <hr />
      <p>
        this is a test
      </p>
    </div>
  </button>
</a>
`;
